src/Qir/qir-utils.ps1

```diff
@@ -34,163 +34,41 @@ function Build-CMakeProject {
-
-
-    # Temporarily disable the following warnings (until QIR RT is refactored to expose C interface).
-
-    # Looks like the `-Wglobal-constructors` warns that the instance of the `__dllexport` class/struct (or a static member var of such class/struct) 
-    # needs to be constructible by calling a global `__dllexport` function (to guarantee that a single instance is created and the same instance is used 
-    # both inside and outside of the binary (dynamic library or executable)).
-    # Or it warns about the constructor that is invoked for a global (or static member) variable _before_ the `main()` is invoked, thus slowing down the start,
-    # see https://stackoverflow.com/a/15708829/6362941
-    $warningFlags += " -Wno-global-constructors"    # https://clang.llvm.org/docs/DiagnosticsReference.html#wglobal-constructors
-    # Looks like the `-Wexit-time-destructors` warns that the destructor of a global or static member variable will be invoked
-    # _after_ the `main()` returns (thus slowing down the termination/restart).
-    $warningFlags += " -Wno-exit-time-destructors"
-
-    # Temporarily disable "-Wextra-semi-stmt" that warns about redundant `;` in the end of `INFO(id);` of Catch tests framework (which looks fixed in the latest Catch version).
-    # Disable until the Catch header "src\Qir\Common\Externals\catch2\catch.hpp" is updated to a version newer than v2.12.1 (from https://github.com/catchorg/Catch2).
-    $warningFlags += " -Wno-extra-semi-stmt"    # https://clang.llvm.org/docs/DiagnosticsReference.html#wextra-semi-stmt
-
-    $env:CFLAGS   += $warningFlags
-    $env:CXXFLAGS += $warningFlags
-
-
-    if ($Env:BUILD_CONFIGURATION -eq "Debug") 
-    { 
+    if ($Env:BUILD_CONFIGURATION -eq "Debug") { 
         # Sanitizers (https://clang.llvm.org/docs/UsersManual.html#controlling-code-generation):
-
-        $sanitizeFlags = ""
-        if (-not ($IsWindows))
-        {
-            # Undefined Behavior Sanitizer (https://clang.llvm.org/docs/UndefinedBehaviorSanitizer.html)
-            # Win:
-            #   FAILED: lib/QIR/Microsoft.Quantum.Qir.Runtime.dll lib/QIR/Microsoft.Quantum.Qir.Runtime.lib
-            #   lld-link: error: /failifmismatch: mismatch detected for 'RuntimeLibrary':
-            #   >>> lib/QIR/CMakeFiles/qir-rt-support-obj.dir/QubitManager.cpp.obj has value MD_DynamicRelease
-            #   >>> clang_rt.ubsan_standalone_cxx-x86_64.lib(ubsan_type_hash_win.cc.obj) has value MT_StaticRelease
-            #   clang++: error: linker command failed with exit code 1 (use -v to see invocation)
-            $sanitizeFlags += " -fsanitize=undefined -fsanitize=float-divide-by-zero -fsanitize=unsigned-integer-overflow -fsanitize=implicit-conversion -fsanitize=local-bounds -fsanitize=nullability"
-            # TODO: 
-            #     For Win consider extra build configuration linking all libs statically, enable `-fsanitize=undefined`, run the statically linked tests.
```

Comment:
I see that after this PR most of the comments (of the sanitizer options) will not survive.
I would strongly recommend to keep the comments. They are a result of a number of hours spent, and they will save from spending those hours again in the future. I'll explain.

Enabling the sanitizers is an incomplete work. At the moment of enabling, certain things were not supported on specific platforms or some sanitizers were conflicting with each other. The sanitizers will continue being developed and some time later we will get back to them and enable more. 
The comments describe certain stage of achievement on our way to enable the sanitizers. The comments say that certain directions are dead ends, etc. If the comments will not survive, then in the future, when we get back to sanitizers, we will have to repeat gaining all that knowledge saved in the comments.